cappuccino/urlinfo.py

```diff
@@ -177,19 +177,21 @@ def _stream_response(self, response: requests.Response) -> str:
         return content.getvalue()
 
     def _process_url(self, url: str):
-        hostname = urlparse(url).hostname
+        urlp = urlparse(url)
+        if urlp.netloc.lower().removeprefix("www.") == "twitter.com":
+            urlp = urlp._replace(netloc="nitter.net")
+
+        hostname = urlp.hostname
         for (_, _, _, _, sockaddr) in socket.getaddrinfo(hostname, None):
             ip = ipaddress.ip_address(sockaddr[0])
             if not ip.is_global:
                 raise InvalidIPAddress(
                     f"{hostname} is not a publicly routable address."
                 )
 
-        hostname = hostname.removeprefix("www.")
-
         # Spoof user agent for certain sites so they give up their secrets.
         request = copy(self.bot.requests)
-        if any(host.endswith(hostname) for host in self._fake_useragent_hostnames):
+        if any(f".{hostname}".endswith(f".{host}") for host in self._fake_useragent_hostnames):
```

Comment:
Did you actually intend for `www.foo.com` to match `foo.com`, instead of the opposite?

Reply:
Yes I did.

Reply:
ok then